include/openssl/hpke.h

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/**
+ * @file
+ * APIs and data structures for HPKE (RFC9180).
+ */
+
+#ifndef OSSL_HPKE_H
+# define OSSL_HPKE_H
+# pragma once
+
+# include <openssl/obj_mac.h>
+# include <openssl/types.h>
+/*
+ * The HPKE modes
+ */
+# define OSSL_HPKE_MODE_BASE              0 /**< Base mode  */
+# define OSSL_HPKE_MODE_PSK               1 /**< Pre-shared key mode */
+# define OSSL_HPKE_MODE_AUTH              2 /**< Authenticated mode */
+# define OSSL_HPKE_MODE_PSKAUTH           3 /**< PSK+authenticated mode */
+
+/*
+ * The (16bit) HPKE algorithn ID IANA codepoints
+ *
+ * If/when new IANA codepoints are added there are tables in
+ * the library implementation that must also be updated.
```

Comment:
Maybe refer to where these come from in the RFC

Reply:
They're spread about the RFC but I can put in a URL for the IANA registries

Reply:
crap - I forgot that - the IANA registtries are at https://www.iana.org/assignments/hpke/hpke.xhtml anyway - I've added that to the comment and it'll be in whatever is the next commit/push